### PR TITLE
[dif/pwm] Add autogen, header, and checklist

### PR DESCRIPTION
**Pull request by timothytrippel**
This adds part of the DIF library for the PWM, including:
1. all autogenerated DIFs,
2. a header with a suggested API,
3. a checklist.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

File: sw/device/lib/dif/dif_pwm.h [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  X(2)                      \
+  X(3)                      \
+  X(4)                      \
+  X(5)
+
+/**
+ * Helper macro for defining a `dif_pwm_channel_t` enumeration constant.
+ * @channel_ PWM channel of the enumeration constant.
+ */
+#define PWM_CHANNEL_ENUM_INIT_(channel_) \
```
Comment (mcy) [inline sw/device/lib/dif/dif_pwm.h:58]: Make sure to undef this.

File: sw/device/lib/dif/dif_pwm.h [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  /**
+   * The PWM duty cycle linearly sweeps between both primary and secondary
+   * firmware-configured values, based on a firmware-configured step size.
+   */
+  kDifPwmModeHeartbeat = 1,
+  /**
+   * The PWM duty cycle alternates between both primary and secondary
+   * firmware-configured values, based on two firmware-configured durations.
+   */
+  kDifPwmModeBlink = 1,
```
Comment (mcy) [inline sw/device/lib/dif/dif_pwm.h:116]: This is a repeat of heartbeat, which I assume is a typo.

File: sw/device/lib/dif/dif_pwm.h [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+   * operation modes.
+   *
+   * The meaning of this parameter is different based on the operation mode:
+   *     - Heartbeat mode: determines the number of "pulse cycles" between
+   *                       increments/decrements to the duty cycle.
+   *
+   *     - Blink mode: determines the number of "pulse cycles" to pulse at duty
+   *                   cycle A, before switching to duty cycle B.
+   */
+  uint16_t blink_parameter_x;
```
Comment (mcy) [inline sw/device/lib/dif/dif_pwm.h:187]: TBH I would have made this into a `u16[2]` to hel make the comment shorter but it's w/e

File: sw/device/lib/dif/dif_pwm.h [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+ * enablement state.
+ *
+ * This function should only need to be called once for the lifetime of
+ * `handle`.
+ *
+ * @param pwm A PWM handle.
+ * @param config Runtime configuration parameters.
+ * @return The result of the operation.
+ */
+OT_WARN_UNUSED_RESULT dif_result_t dif_pwm_configure(const dif_pwm_t *pwm,
```
Comment (mcy) [inline sw/device/lib/dif/dif_pwm.h:222]: Missing newline after the attr.

**Review by mcy — APPROVED** [review]

File: sw/device/lib/dif/dif_pwm.h (hunk 1 above)
Comment (timothytrippel) [reply sw/device/lib/dif/dif_pwm.h:58]: Good catch, fixed.

File: sw/device/lib/dif/dif_pwm.h (hunk 2 above)
Comment (timothytrippel) [reply sw/device/lib/dif/dif_pwm.h:116]: Fixed.

File: sw/device/lib/dif/dif_pwm.h (hunk 4 above)
Comment (timothytrippel) [reply sw/device/lib/dif/dif_pwm.h:222]: Fixed.

**Pull request merged**